app/api/models/errors/CommonMtdErrors.scala

```diff
@@ -14,84 +14,77 @@
+    extends MtdError("RULE_INCORRECT_OR_EMPTY_BODY_SUBMITTED", "An empty or non-matching body was submitted", BAD_REQUEST)
 
-object RuleDeleteForbiddenError extends MtdError("RULE_DELETE_FORBIDDEN", "A deletion of a HMRC held state benefit is not permitted")
+object RuleDeleteForbiddenError extends MtdError("RULE_DELETE_FORBIDDEN", "A deletion of a HMRC held state benefit is not permitted", BAD_REQUEST)
 
-object RuleUpdateForbiddenError extends MtdError("RULE_UPDATE_FORBIDDEN", "The update for a HMRC held benefit is not permitted")
+object RuleUpdateForbiddenError extends MtdError("RULE_UPDATE_FORBIDDEN", "The update for a HMRC held benefit is not permitted", BAD_REQUEST)
 
-object RuleIgnoreForbiddenError extends MtdError("RULE_IGNORE_FORBIDDEN", "A customer added state benefit cannot be ignored")
+object RuleIgnoreForbiddenError extends MtdError("RULE_IGNORE_FORBIDDEN", "A customer added state benefit cannot be ignored", BAD_REQUEST)
 
-object RuleUnignoreForbiddenError extends MtdError("RULE_UNIGNORE_FORBIDDEN", "A customer added state benefit cannot be unignored")
+object RuleUnignoreForbiddenError extends MtdError("RULE_UNIGNORE_FORBIDDEN", "A customer added state benefit cannot be unignored", BAD_REQUEST)
 
-object RuleBenefitTypeExists extends MtdError("RULE_BENEFIT_TYPE_EXISTS", "A benefit of this type has already been created")
+object RuleBenefitTypeExists extends MtdError("RULE_BENEFIT_TYPE_EXISTS", "A benefit of this type has already been created", BAD_REQUEST)
 
 //Standard Errors
-object NotFoundError extends MtdError("MATCHING_RESOURCE_NOT_FOUND", "Matching resource not found")
+object NotFoundError extends MtdError("MATCHING_RESOURCE_NOT_FOUND", "Matching resource not found", NOT_FOUND)
 
-object StandardDownstreamError extends MtdError("INTERNAL_SERVER_ERROR", "An internal server error occurred")
+object StandardDownstreamError extends MtdError("INTERNAL_SERVER_ERROR", "An internal server error occurred", INTERNAL_SERVER_ERROR)
 
-object BadRequestError extends MtdError("INVALID_REQUEST", "Invalid request")
+object BadRequestError extends MtdError("INVALID_REQUEST", "Invalid request", BAD_REQUEST)
 
-object BVRError extends MtdError("BUSINESS_ERROR", "Business validation error")
+object BVRError extends MtdError("BUSINESS_ERROR", "Business validation error", BAD_REQUEST)
 
-object ServiceUnavailableError extends MtdError("SERVICE_UNAVAILABLE", "Internal server error")
+object ServiceUnavailableError extends MtdError("SERVICE_UNAVAILABLE", "Internal server error", INTERNAL_SERVER_ERROR)
 
 //Authorisation Errors
-object UnauthorisedError extends MtdError("CLIENT_OR_AGENT_NOT_AUTHORISED", "The client or agent is not authorised")
+object ClientNotAuthorisedError extends MtdError("CLIENT_OR_AGENT_NOT_AUTHORISED", "The client or agent is not authorised", FORBIDDEN)
+
+object ClientNotAuthenticatedError extends MtdError("CLIENT_OR_AGENT_NOT_AUTHORISED", "The client or agent is not authorised", UNAUTHORIZED)
```

Comment:
How come the error name is not _authenticated_ but the code, message and status are not _authorised_

Reply:
These errors have the same error message, just different error codes. Again, part of the common code changes